packages/plugin-core/src/test/MockDendronExtension.ts

```diff
@@ -0,0 +1,170 @@
+    return this._context;
+  }
+
+  serverWatcher?: FileSystemWatcher | undefined;
+
+  get type(): WorkspaceType {
+    throw new Error("Method not implemented in MockDendronExtension");
+  }
+  get wsUtils(): IWSUtilsV2 {
+    return new WSUtilsV2(this);
+  }
+  get commandFactory(): ICommandFactory {
+    throw new Error("Method not implemented in MockDendronExtension");
+  }
+  get schemaSyncService(): ISchemaSyncService {
+    throw new Error("Method not implemented in MockDendronExtension");
+  }
+  get workspaceService(): IWorkspaceService | undefined {
+    throw new Error("Method not implemented in MockDendronExtension");
+  }
+  get noteSyncService(): INoteSyncService {
+    throw new Error("Method not implemented in MockDendronExtension");
+  }
+  get lookupControllerFactory(): ILookupControllerV3Factory {
+    throw new Error("Method not implemented in MockDendronExtension");
+  }
+  get noteLookupProviderFactory(): INoteLookupProviderFactory {
+    throw new Error("Method not implemented in MockDendronExtension");
+  }
+
+  get schemaLookupProviderFactory(): ISchemaLookupProviderFactory {
+    throw new Error("Method not implemented in MockDendronExtension.");
+  }
+
+  /**
+   * Note: No-Op
+   * @param _cb
+   * @returns
+   */
+  pauseWatchers<T = void>(cb: () => Promise<T>): Promise<T> {
+    return cb();
+  }
+
+  getClientAPIRootUrl(): Promise<string> {
+    throw new Error("Method not implemented in MockDendronExtension.");
+  }
+  getDWorkspace(): DWorkspaceV2 {
+    const ret: any = {
+      wsRoot: this._wsRoot,
+      vaults: this._vaults,
+    };
+
+    return ret;
+  }
+  getWorkspaceImplOrThrow(): DWorkspaceV2 {
+    throw new Error("Method not implemented in MockDendronExtension.");
+  }
+  getWorkspaceSettings(): Promise<WorkspaceSettings | undefined> {
+    throw new Error("Method not implemented in MockDendronExtension.");
+  }
+  getWorkspaceSettingsSync(): WorkspaceSettings | undefined {
+    throw new Error("Method not implemented in MockDendronExtension.");
+  }
+  getDendronWorkspaceSettingsSync():
+    | Partial<{
+        "dendron.dailyJournalDomain": string;
+        "dendron.defaultJournalName": string;
+        "dendron.defaultJournalDateFormat": string;
+        "dendron.defaultJournalAddBehavior": string;
+        "dendron.defaultScratchName": string;
+        "dendron.defaultScratchDateFormat": string;
+        "dendron.defaultScratchAddBehavior": string;
+        "dendron.copyNoteUrlRoot": string;
+        "dendron.linkSelectAutoTitleBehavior": string;
+        "dendron.defaultLookupCreateBehavior": string;
+        "dendron.defaultTimestampDecorationFormat": string;
+        "dendron.rootDir": string;
+        "dendron.dendronDir": string;
+        "dendron.logLevel": string;
+        "dendron.trace.server": string;
+        "dendron.serverPort": string;
+      }>
+    | undefined {
+    throw new Error("Method not implemented in MockDendronExtension.");
+  }
+  getWorkspaceSettingOrDefault({}: {
```

Comment:
can you make sure `eslint` is enabled in your workspace by downloading and enabling the [eslint extension](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint)